include/tvSettings.h

```diff
@@ -2567,6 +2567,62 @@ tvError_t EnableDynamicContrast(bool mode);
  */
 tvError_t EnableLocalContrast(bool mode);
 
+/** 
+* @brief Triggers short circuit detection operation and returns the status.
+*
+* On calling GetLdimZoneShortCircuitStatus, the short circuit detection logic
+* is triggered. It detects short circuits between zones.  On completion, it
+* returns a list of zones where a short circuit was detected.
+*
+* Shall return OPERATION_NOT_SUPPORTED, if this feature is not supported
+* on the called platform.
+*
+*
+* @param[out] status
+*     - int*, Callee-saved status indicating short circuit detected or not.
+*     - 0 indicates success, no short circuit detected.
+*     - 1 indicates failure, at-least one short circuit detected.
+*     - This should be pre-allocated by the caller.
+*
+* @param[out] shortcircuit_zone_list
+*     - unsigned char*, each array element represents a zone short circuit status,
+*       like, [0][1][1][0]...[0] represents short detected or not.
+*       0 - no short circuit detected for the respective zone.
+*       1 - short circuit detected for the respective zone.
+*     - On input, this is an allocated array; on output, each indices
+*       represents zone number and it's value shows status of short
+*       detected/ not-detected status (on success).
+*     - This array should be pre-allocated by the caller with sufficient
+*       size to hold the maximum number of zones.
+*
+* @param[in] size
+*     - int, the size of 'shortcircuit_zone_list' array
+*
+* @return tvError_t
+*
+* @retval tvERROR_NONE
```

Comment:
not the same format as the other API's.

Reply:
I'm not sure what does this mean ?  Is this about the way white space characters are used ?

Is there a DTD definition available to follow ?

Reply:
```c
/**
 * @brief Triggers short circuit detection and returns the status.
 *
 * This function initiates the short circuit detection process between zones. Upon completion, 
 * it populates an array indicating which zones have a short circuit.
 *
 * @param[out] shortcircuit_zone_list Array where each element represents a zone's short circuit status.
 *                                   0: No short circuit detected for the zone.
 *                                   1: Short circuit detected for the zone.
 *                                   This array must be pre-allocated by the caller with sufficient size.
 * @param[in]  size                 The size of the `shortcircuit_zone_list` array.
 * @param[out] status               Pointer to a variable to store the overall status.
 *                                   0: Success, no short circuit detected.
 *                                   1: Failure, at least one short circuit detected.
 *                                   This variable must be pre-allocated by the caller.
 *
 * @returns tvERROR_NONE             Short circuit detection completed successfully.
 * @returns tvERROR_INVALID_PARAM    Invalid input parameter.
 * @returns tvERROR_INVALID_STATE    Detection logic failed to execute.
 * @returns tvERROR_GENERAL          Underlying failure (e.g., SoC, memory).
 * @returns tvERROR_OPERATION_NOT_SUPPORTED Operation not supported on this platform.
 *
 * @pre TvInit() must be called before calling this API.
 */
tvError_t GetLdimZoneShortCircuitStatus(unsigned char* shortcircuit_zone_list, int size, int* status);
```

Reply:
the issue of course is that all the error codes needs to be tested.

so `tvERROR_GENERAL` you should be explicit if possible `tvERROR_OUT_OF_MEMORY` can be tested, but not a GENERAL error across all vendors?

so basically I'm saying 

will - `tvERROR_INVALID_STATE` & `tvERROR_GENERAL` really occur? can we remove them?

Reply:
without `-`

```c
/**
 * @brief Triggers short circuit detection and returns the status.
 *
 * On calling GetLdimZoneShortCircuitStatus, the short circuit detection logic
 * is triggered. It detects short circuits between zones. On completion, it
 * returns a list of zones where a short circuit was detected.
 *
 * @param[out] pulShortcircuitZoneList  Array indicating short circuit status for each zone.
 *                                      0 - No short circuit detected.
 *                                      1 - Short circuit detected.
 * @param[in]  ulSize                   Size of the pulShortcircuitZoneList array.
 * @param[out] pStatus                  Short circuit detection status.
 *                                      0 - Success, no short circuit detected.
 *                                      1 - Failure, at least one short circuit detected.
 *
 * @return tvError_t
 * @retval tvERROR_NONE                   Short circuit detection completed successfully.
 * @retval tvERROR_INVALID_PARAM          Invalid input parameter.
 * @retval tvERROR_INVALID_STATE          Detection logic failed to execute.
 * @retval tvERROR_GENERAL                Underlying failures (SoC, memory, etc.).
 * @retval tvERROR_OPERATION_NOT_SUPPORTED Operation is not supported on this platform.
 *
 * @pre TvInit() should be called before calling this API.
 */
tvError_t GetLdimZoneShortCircuitStatus(unsigned char* pulShortcircuitZoneList, int ulSize, int* pStatus);
```

Reply:
tvERROR_INVALID_STATE, is reported when HAL is not in a state to trigger the test.  Currently HAL impl returns invalid state when its resource handles are not valid.

tvERROR_GENERAL, is reported if any error from drivers/soc